core/eth/contracts.go

```diff
@@ -86,9 +101,158 @@ func MustGetV6ContractEventID(name, eventName string) common.Hash {
 		logger.Panic(fmt.Errorf("unable to find contract %s", name))
 	}
 
-	event, found := contract.ABI.Events[eventName]
+	event, found := contract.ABI().Events[eventName]
 	if !found {
 		logger.Panic(fmt.Errorf("unable to find event %s for contract %s", eventName, name))
 	}
 	return event.ID()
 }
+
```

Comment:
`UnpackLog` and `parseTopics` are taken verbatim from the Geth codebase, which doesn't expose them in a way we can use.

Reply:
Maybe comment on their origin and reason for import?  we can carry the info from this PR into the codebas

Reply:
naive suggestion: what do you think about isolating the code carried from geth into its own file?  Or having a special label for it (in the comments)?

Reply:
@felder-cl I like the idea.  I think there are a couple of other places we might be doing this as well, and it'd be nice to have everything "vendored" in a single location.

@se3000  Yeah, `BoundContract` needs to be handed something implementing a number of Geth's interfaces that our `eth.Client` doesn't currently implement, and Geth's `ethclient.Client` needs to be handed a concrete `*rpc.Client` (rather than an interface), which conflicts with the fact that we wrap `rpc.Client` in `lazyRPCWrapper`.  My intention long-term is to try to get us using more of Geth's built in types and code generation facilities, but for now, it seemed out of scope.